### PR TITLE
Fix LTI 1.3 grade timestamps

### PR DESCRIPTION
**Pull request by trombonekenny**
- Changes to always send the graded timestamp as the current time so that Send Grades updates Canvas regardless of previous grades
- Fixes modal "Pick from existing assignments" to refresh a single div instead of repeating, added a refresh button
- Includes documentation about how to edit an existing Canvas assignment to show up in the line item list
- Adds error message display to the grade sync for the instructor
- Add instructions to the instructor LTI 1.3 admin page on how to refresh LTI metadata rather than erroring

### CODE REVIEW

**Comment by trombonekenny** [comment]
image

image

**Comment by trombonekenny** [comment]
> There are now four differently-colored buttons and a link in this modal. This is getting crazy. We do not need so many colors. I won't necessarily block on this, but if you merge this, let's pause after that and do some UI work.

It's easy enough to change the colors while the PR is open. I was working off of our old conventions: green makes something new, blue updates/edits, and grey cancels. I don't think color should be the only sense of context, but I was trying to be consistent and recognizable (more than necessarily colorful).

How would you have me simplify this? I can take out the Refresh button and add the icon to the "Pick from existing..." button to make it little less cluttered. Switching to details/summary for the extra info might help too?

File: apps/prairielearn/src/ee/pages/instructorInstanceAdminLti13/instructorInstanceAdminLti13.html.ts [hunk 1]
```diff
@@ -350,6 +138,211 @@ export function InstructorInstanceAdminLti13({
...
+                        </button>
+                        <span class="dropdown js-question-actions">
+                          <button
+                            type="button"
+                            class="btn btn-xs btn-light dropdown-toggle"
+                            data-toggle="dropdown"
+                            aria-haspopup="true"
+                            aria-expanded="false"
+                          >
+                            ...<span class="caret"></span>
```
[the remark this answers is not in the document]
Comment (trombonekenny) [reply apps/prairielearn/src/ee/pages/instructorInstanceAdminLti13/instructorInstanceAdminLti13.html.ts:321]: I'm not sure; I must have copied it from somewhere. I kind of like the split button approach at the page you mentioned. Does this work?

image

[the remark this answers is not in the document]
Comment (nwalters512) [reply apps/prairielearn/src/ee/pages/instructorInstanceAdminLti13/instructorInstanceAdminLti13.html.ts:321]: Works for me, yeah!

**Comment by nwalters512** [comment]
> I was working off of our old conventions: green makes something new, blue updates/edits, and grey cancels.

I'm reasonably sure that you and you alone invented the "green makes something new" convention over the course of the past year. Candidly: I had no idea that why you chose to use green buttons in some situations until you spelled it out for me just now, and I'm really not sure it's a useful distinction. (I went and checked `git blame`, and 8 out of the 12 `btn-success` buttons that appear in normal production usage are from you in the past year. I share this only because I was curious about the origin of these green buttons.)

I'd be happy if our palette was predominantly blue (`primary`), gray (`secondary`), with red (`danger`) for destructive actions when needed. I tend to think that we should only use teal (`info`) when we _really_ need another color, and in most cases I don't think we do.

I used https://v0.dev/chat to play around with some ideas and I kind of like some of the things that are going on in this one: https://b_bi6veckjj4d.v0.build/

Screenshot 2024-10-07 at 16 39 12

Things I like here:

- The UI for "create new" and "select existing" are fully separate
- The primary affirmative action button (either "Create and link" or "Link existing") is in the conventional spot in the modal
- There's an option to customize the name before creation

I'm definitely not focused on the particular visual style here, we'd want to adapt it to our own look and feel and the Bootstrap UI elements. Maybe something like tabs for the "create"/"select" choice?

Screenshot 2024-10-07 at 16 41 40

I think using radio buttons (like you already are) for picking existing assignments is fine.

I'm happy to play around with UI choices on your behalf, I enjoy this kind of work. Also happy to review or chat about ideas you may have.

**Review by nwalters512 — APPROVED** [review]
This all looks reasonable. We can save UI tweaks for a future PR.

**Pull request merged**